tests/contenttypes_tests/test_checks.py

```diff
@@ -216,3 +217,29 @@ class InvalidBookmark(models.Model):
                 id='fields.E001',
             )
         ])
+
+
+@isolate_apps('contenttypes_tests', attr_name='apps')
+class ModelCheckTests(SimpleTestCase):
+    def test_model_name_too_long(self):
+        class ModelWithNameOfOver100CharactersModelWithNameOfOver100CharactersModelWithNameOfOver100CharactersModel(
```

Comment:
I wonder if this might be bit more readable:
`model = type('A' * 101, (models.Model,), {'__module__': self.__module__})`